paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentSheetLoader.kt

```diff
@@ -104,6 +104,10 @@ internal class DefaultPaymentSheetLoader @Inject constructor(
         }?.isReady()?.first() ?: false
     }
 
+    private suspend fun isGooglePaySupported(): Boolean {
+        return googlePayRepositoryFactory(GooglePayEnvironment.Test).isReady().first()
+    }
```

Comment:
I haven't been a better solution for checking if Google Pay is supported other than [isReadyToPay](https://developers.google.com/pay/api/android/reference/client#isReadyToPay). The android docs doesn't make it clear the differences of using a `production` versus `test` environment but the [web docs](https://developers.google.com/pay/api/web/reference/client#isReadyToPay) do indicate that a `paymentMethodPresent` will always be true. Should we initialize this as a production environment to ensure we don't get any unexpected results?

Reply:
@samer-stripe could you dynamically set the environment based configuration somehow?

Reply:
@samer-stripe Should the `GooglePayEnvironment` not be set based on the corresponding value in `PaymentSheet.Configuration`?

Reply:
In this scenario, we are only checking if the device supports Google Pay, regardless of merchant settings. This is only accurately done through `Production` environment. iOS does the same thing for Apple Pay.